src/raw.rs

```diff
@@ -11,10 +11,294 @@
+    /// # let connected_client = connecting_client.wait().unwrap();
+    /// let kvpair1 = ("PD", "Go");
+    /// let kvpair2 = ("TiKV", "Rust");
+    /// let iterable = vec![kvpair1, kvpair2];
+    /// let req = connected_client.batch_put(iterable);
+    /// let result: () = req.wait().unwrap();
+    /// ```
+    pub fn batch_put(&self, pairs: impl IntoIterator<Item = impl Into<KvPair>>) -> BatchPut {
+        BatchPut::new(
+            self.rpc(),
+            BatchPutInner::new(pairs.into_iter().map(Into::into).collect()),
+        )
+    }
+
+    /// Create a new [`Delete`](struct.Delete.html) request.
+    ///
+    /// Once resolved this request will result in the deletion of the given key.
+    ///
+    /// ```rust,no_run
+    /// # use tikv_client::{Key, Config, raw::Client};
+    /// # use futures::Future;
+    /// # let connecting_client = Client::new(&Config::new(vec!["192.168.0.100", "192.168.0.101"]));
+    /// # let connected_client = connecting_client.wait().unwrap();
+    /// let key = "TiKV";
+    /// let req = connected_client.delete(key);
+    /// let result: () = req.wait().unwrap();
+    /// ```
+    pub fn delete(&self, key: impl Into<Key>) -> Delete {
+        Delete::new(self.rpc(), DeleteInner::new(key.into()))
+    }
+
+    /// Create a new [`BatchDelete`](struct.BatchDelete.html) request.
+    ///
+    /// Once resolved this request will result in the deletion of the given keys.
+    ///
+    /// ```rust,no_run
+    /// # use tikv_client::{Config, raw::Client};
+    /// # use futures::Future;
+    /// # let connecting_client = Client::new(&Config::new(vec!["192.168.0.100", "192.168.0.101"]));
+    /// # let connected_client = connecting_client.wait().unwrap();
+    /// let keys = vec!["TiKV", "TiDB"];
+    /// let req = connected_client.batch_delete(keys);
+    /// let result: () = req.wait().unwrap();
+    /// ```
+    pub fn batch_delete(&self, keys: impl IntoIterator<Item = impl Into<Key>>) -> BatchDelete {
+        BatchDelete::new(
+            self.rpc(),
+            BatchDeleteInner::new(keys.into_iter().map(Into::into).collect()),
+        )
+    }
+
+    /// Create a new [`Scan`](struct.Scan.html) request.
+    ///
+    /// Once resolved this request will result in a scanner over the given keys.
+    ///
+    /// If not passed a `limit` parameter, it will default to `u32::MAX`.
+    ///
+    /// ```rust,no_run
+    /// # use tikv_client::{KvPair, Config, raw::Client};
+    /// # use futures::Future;
+    /// # let connecting_client = Client::new(&Config::new(vec!["192.168.0.100", "192.168.0.101"]));
+    /// # let connected_client = connecting_client.wait().unwrap();
+    /// let inclusive_range = "TiKV"..="TiDB";
+    /// let req = connected_client.scan(inclusive_range, 2);
+    /// let result: Vec<KvPair> = req.wait().unwrap();
+    /// ```
+    pub fn scan(&self, range: impl KeyRange, limit: impl Into<Option<u32>>) -> Scan {
```

Comment:
BTW: The max limit can be changed in Golang client as `MaxRawKVScanLimit` is a public variable. I'm not sure if we want to let user change this via Config or simply do not allow user to change it. I'm using a const right now and would like to hear comments from everybody about this.

Reply:
I'm not even sure what this limit is, but I suspect I prefer giving users the option to not think about it, so either Option or Into<Option>. I don't have strong feelings about which one.

I think I don't know the domain well enough yet to have an opinion on exactly where it should be possible to set this value.

Reply:
Since the underlying RPC protocol requires limit to be set, and there should be a rational upper bound. I suggest changing this argument to required. @Hoverbear @siddontang What do you think?

Reply:
@sunxiaoguang I think that's fine. Let's make sure to expose a 'max'/easy const as `tikv_client::NO_SCAN_LIMIT` or something so users can find it easily.